perun-core/src/test/java/cz/metacentrum/perun/core/api/AuthzResolverIntegrationTest.java

```diff
@@ -1660,6 +1663,114 @@ public void mfaNotCriticalObject() throws Exception {
 		}
 	}
 
+	@Test
+	public void mfaGenericSetRole() throws Exception {
+		System.out.println(CLASS_NAME + "mfaGenericSetRole");
+
+		final Vo createdVo = perun.getVosManager().createVo(sess, new Vo(0,"test123test123","test123test123"));
+		final Member createdMember = createSomeMember(createdVo);
+		final User createdUser = perun.getUsersManagerBl().getUserByMember(sess, createdMember);
+
+		boolean originalMfaForce = BeansUtils.getCoreConfig().isEnforceMfa();
```

Comment:
Maybe set assignmentCheck for PERUNADMIN as you did for VOADMIN in the following tests? Just to be consistent and extra safe. What do you think?